docs/HOW_TO_OVERRIDE_STYLES_PROPERLY.md

```diff
@@ -0,0 +1,144 @@
+}
+```
+These variables will be found in the `Style API` tab in the component's storybook.
+
+If the component uses the new approach for overrides (using the `overrideStyleParams` class), 
+use that to extend the component (as described in the aforementioned [guide](
+https://github.com/wix/wix-ui-tpa/blob/master/docs/USAGE.md#an-optimized-way-for-style-overriding---new))  
+If not, then use its "default" `.root` export for overrides.
+
+In any case **DO NOT** use `-st-extends` to override styles by targeting internal elements.
+
+#### Examples
+
+✅ DO
+
+```css
+:import {
+   -st-from: "wix-ui-tpa/.../Component.st.css";
+   -st-named: overrideStyleParams;
+   
+   /* If overrideStyleParams doesn't exist use: */
+   /* -st-default: TPAComponent; */
+}
+
+.myComponent {
+   -st-mixin: overrideStyleParams(
+     TextColor <your-color-here>
+   );
+
+   /* If overrideStyleParams doesn't exist use: */
+   /* -st-mixin: TPAComponent(...); */
+}
+```
+
+❌ DON'T
+
+```css
+:import {
+   -st-from: "wix-ui-tpa/.../Component.st.css";
+   -st-default: TPAComponent; 
+}
+
+.myComponent {
+   -st-extends: TPAComponent;
+}
+
+.myComponent::innerElement {
+   background-color: <some-color>;
+}
+```
+
+## FAQ's
+
+#### "The component doesn't expose the variables I need, what should I do?"
+First, contact us at [#wix-ui-tpa](slack://channel?id=CJRU3U97A&team=T02T01M9Y).  
+We'll check with our designers that the variables align with the [design system](
+https://zeroheight.com/7sjjzhgo2/p/7181b5-tpa-design-system).  
+If the designers approve, you'll be able to contribute the new variables.
+
+#### "Can I style the component's root element?"
+Well yes, and no 😳  
+All of our components accept a `className` prop, which is applied to the root element.  
+In addition to extending styles, using `-st-mixin`, you can use this className to control the 
+layout of you app.  
+So you can use rules such as `position`, `width`, `left`, `top`, `flex`, etc.  
+Don't use this className to change text or background color, border's or padding, as this might 
+break the component, or it might not be aligned to the design system at all.  
+Also beware of using the `display` rule on the root element, as this may break the component's 
+internal layout. If you need to change its `display` and it breaks the component, please contact us
+at our [slack channel](slack://channel?id=CJRU3U97A&team=T02T01M9Y) and we'll try to find a solution 
+together.
+
+Examples:
+
+✅ DO
+
+```css
+:import {
+   -st-from: "wix-ui-tpa/.../Component.st.css";
+   -st-named: overrideStyleParams;
+}
+
+.myComponent {
+   -st-mixin: overrideStyleParams(...);
+   position: absolute;
+   top: 0;
+   left: 0;
+}
+```
+
+❌ DON'T
+
+```css
+:import {
+   -st-from: "wix-ui-tpa/.../Component.st.css";
+   -st-named: overrideStyleParams;
+}
+
+.myComponent {
+   -st-mixin: overrideStyleParams(...);
+   background-color: cornflowerblue;
+   border: 1px solid gold;
+   display: inline;
+}
```

Comment:
Also here, I would use some comments to point the important stuff - not sure everyone will read/understand the paragraph above or it might take a little time to detect the reference in the snippet, so just to ease on the readers